src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        return switch (random.nextInt(NUMBER_OF_FIGURES)) {
+            case 1 -> new Square(random.nextInt(), getRandomColor());
+            case 2 -> new Rectangle(random.nextInt(), random.nextInt(), getRandomColor());
+            case 3 -> new RightTriangle(random.nextInt(), random.nextInt(), getRandomColor());
+            case 4 -> new Circle(random.nextInt(), getRandomColor());
+            default -> new IsoscelesTrapezoid(
+                    random.nextInt(), random.nextInt(), random.nextInt(), getRandomColor());
+        };
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, "white");
```

Comment:
All magic numbers in your code should be constants.
```suggestion
        return new Circle(DEFAULT_COLOR, Color.WHITE.name());
```